### PR TITLE
Update space cards UI

### PR DESCRIPTION
**Pull request by alexey-petrov**
Closes:
- https://github.com/sparkletown/internal-sparkle-issues/issues/1468

Space cards UI overhaul

Default background remains black (if custom is not set)

Max number of cards per row - 3
@media breakouts are now displaying 2 and 1 card instead of 3 and 2 ( necessary to fit max length venue name - 50 characters)

Before:
image

image

After:
Max number of cards in a row is now 3
image

Empty Icon is replaced with a default now:
image

### CODE REVIEW

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 1]
```diff
@@ -21,28 +24,71 @@ export interface AdminSpaceCardProps {
...
+            {"Visit"}
+            <FontAwesomeIcon
+              className="AdminSpaceCard__link-icon"
+              icon={faExternalLinkAlt}
+            />
+          </Link>
+          <div className="AdminSpaceCard__body">
+            <div className={logoClasses} />
+            <div className="AdminSpaceCard__body-info">
+              <AdminShowcaseSubTitle>{venue.name}</AdminShowcaseSubTitle>
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:54]: Maybe not the proper component for this place. Can't hurt to make a copy of it, name it `CardSubTitle`, add the proper SCSS there, and never need bother overriding the `AdminShowcaseSubTitle` style

Comment (goran-peoski-work) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:54]: Well, considering I don't see `AdminShowcaseTitle` here... can be named `CardTitle` instead 🤷

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 2]
```diff
@@ -21,28 +24,71 @@ export interface AdminSpaceCardProps {
...
+              <span className="AdminSpaceCard__text">
+                <img
+                  alt={`space-icon-${spaceIcon}`}
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:57]: Rule of thumb: 

alt text should read in plain English exactly like you'd want some one to explain it to you on the phone.

```suggestion
                  alt={`the ${SPACE_TAXON.lower} icon`}
```

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 3]
```diff
@@ -21,28 +24,71 @@ export interface AdminSpaceCardProps {
...
+            </div>
+          )}
+        </div>
+      </div>
+      <div className="AdminSpaceCard__footer">
+        <div className="AdminSpaceCard__footer-content">
+          <div className="AdminSpaceCard__description">
+            <span className="AdminSpaceCard__description-text">
+              {venue.config?.landingPageConfig?.description ||
+                "Description can be changed in space settings"}
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:84]: Maybe a variable before `return` that makes this defaulting clearer, not stuck after 4 lines of class names

File: src/pages/SpacesDashboard/SpacesDashboard.scss [hunk 4]
```diff
@@ -26,15 +26,15 @@
...
-    @media (max-width: $media-breakpoint--lg) {
-      grid-template-columns: repeat(4, 1fr);
+    @media (max-width: $media-breakpoint--xxl) {
+      grid-template-columns: repeat(2, 1fr);
     }
 
-    @media (max-width: $media-breakpoint--md) {
+    @media (max-width: $media-breakpoint--xl) {
       grid-template-columns: repeat(1, 1fr);
     }
```
Comment (goran-peoski-work) [inline src/pages/SpacesDashboard/SpacesDashboard.scss:39]: 🚀  Noice. And what happens on wider screens? Do cards expand or does just the space between them?

File: src/scss/constants.scss [hunk 5]
```diff
@@ -139,6 +139,7 @@ $media-breakpoint--sm: 576px;
 $media-breakpoint--md: 768px;
 $media-breakpoint--lg: 992px;
 $media-breakpoint--xl: 1200px;
+$media-breakpoint--xxl: 1350px;
```
Comment (goran-peoski-work) [inline src/scss/constants.scss:142]: Could be useful to add line comments after the `;` on each one of these that adds context. 

Something from these browser lists should do it.
sparkle-display-sizes-02
sparkle-display-sizes-01

File: src/scss/constants.scss [hunk 6]
```diff
@@ -269,6 +271,8 @@ $z-layers: (
   map-grid: 3,
   map-room--unclickable: 2,
   map-background: 1,
+  // Admin Space Card
+  space-card-link: $z-layer-space-card-link,
```
Comment (goran-peoski-work) [inline src/scss/constants.scss:275]: Why is this even necessary? Are there cases of the cards being overlapped or overlapping some other element? Schedule? Chat?

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Code looks good. 
Have mostly puzzling questions since I still didn't run it locally. 
One or two things can be improved.

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 7]
```diff
@@ -1,7 +1,10 @@
 @import "scss/constants.scss";
 
 $admin-space-card-button-margin: $spacing--xs 0;
-$admin-space-card-logo-size: 80px;
+$admin-space-card-logo-size: 64px;
+$admin-space-card-body-height: 80px;
+$admin-space-card-count-top-margin: 120px;
+$admin-space-card-bg-height: 150px;
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss:7]: looks fine on both my laptop/pc
telegram-cloud-photo-size-2-5262570153470704243-y
image
and it holds up pretty nice on resize as well
image

btw I think we should do something about that buttons container, it seems to have a fixed width that doesn't resize along with a window and probably we need to move it above the cards or something

File: src/pages/SpacesDashboard/SpacesDashboard.scss (hunk 4 above)
Comment (alexey-petrov) [reply src/pages/SpacesDashboard/SpacesDashboard.scss:39]: refer to https://github.com/sparkletown/sparkle/pull/2598#discussion_r757473693

cards resize just fine while spaces stay the same, I've tried to play around with different resolutions and didn't find any issues with cards so far

File: src/scss/constants.scss (hunk 6 above)
Comment (alexey-petrov) [reply src/scss/constants.scss:275]: i've added that z-index since I had to put a negative margin to the card's body container, so it fits on different screen resolutions with a max length name in it, no other real reason to have it

File: src/scss/constants.scss (hunk 5 above)
Comment (alexey-petrov) [reply src/scss/constants.scss:147]: wasn't sure if something from that list would work, so I added general namings for those constants, like `tablet`, `mobile`, `tablet-small` etc. f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 3 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: done f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 2 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: done f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 1 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: yeah, you're right
did that f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 8]
```diff
@@ -21,28 +24,71 @@ export interface AdminSpaceCardProps {
...
+
   return (
     <div className="AdminSpaceCard">
-      <div className={backgroundClasses} />
-      <div className="AdminSpaceCard__info">
-        <div className={logoClasses} />
-        <AdminShowcaseSubTitle>{venue.name}</AdminShowcaseSubTitle>
+      <div className={backgroundClasses}>
+        <div className="AdminSpaceCard__bg-container">
+          <Link className="AdminSpaceCard__link" to="">
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: fixed f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 9]
```diff
@@ -21,28 +24,71 @@ export interface AdminSpaceCardProps {
 
 export const AdminSpaceCard: React.FC<AdminSpaceCardProps> = ({ venue }) => {
   const backgroundStyle = useCss({
-    "background-image": `url(${venue.mapBackgroundImageUrl})`,
+    background: `linear-gradient(
+        rgba(25, 24, 26, 0.8),
+        rgba(25, 24, 26, 0.8)
+      ), url(${venue.config?.landingPageConfig?.bannerImageUrl})`,
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: fixed by adding `const [validBannerImageUrl] = useValidImage(
    venue?.config?.landingPageConfig.bannerImageUrl,
    DEFAULT_VENUE_BANNER_COLOR
  );`

f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 10]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
...
   }
 
   .ButtonNG {
     margin: $admin-space-card-button-margin;
   }
 }
+
+@media (max-width: $media-breakpoint--xxl) and (min-width: $media-breakpoint--lg) {
+  .AdminSpaceCard__logo {
+    @include square-size(48px);
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: done f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 11]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
...
   }
 
   &__logo {
     @include square-size($admin-space-card-logo-size);
+    margin-top: $spacing--lg;
     border-radius: $border-radius--max;
     background-size: cover;
     background-color: $content--under;
-    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.4);
+    box-shadow: 0 $spacing--xs $spacing--md rgba(0, 0, 0, 0.4);
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: done f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 12]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
...
     display: flex;
     justify-content: space-between;
+    align-items: center;
     flex-direction: column;
-    z-index: z(admin-space-card-info);
+    flex: 1;
+  }
+
+  &__description {
+    min-height: 100px;
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: fixed f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 13]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
...
-      top: 0;
-      left: 0;
-      width: 100%;
-      height: 100%;
-      background-image: linear-gradient(
-        180deg,
-        rgba(25, 24, 26, 0.6) 0%,
-        $secondary--dark 100%
-      );
+    &-container {
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: fixed f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 14]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
...
+  max-height: 450px;
+  box-shadow: 0px 2px 4px rgba(58, 92, 144, 0.14),
+    0px 3px 4px rgba(58, 92, 144, 0.12), 0px 1px 5px rgba(58, 92, 144, 0.2);
 
   .AdminShowcaseSubTitle {
     font-weight: $font-weight--700;
-    font-size: font-size--parent(1.6);
+    font-size: $font-size--lg;
     color: $content--over;
+    margin: $spacing--md $spacing--md $spacing--sm;
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: removed f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 15]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
...
-  padding: font-size--root(1);
   gap: font-size--root(0.5);
+  max-height: 450px;
+  box-shadow: 0px 2px 4px rgba(58, 92, 144, 0.14),
+    0px 3px 4px rgba(58, 92, 144, 0.12), 0px 1px 5px rgba(58, 92, 144, 0.2);
 
   .AdminShowcaseSubTitle {
     font-weight: $font-weight--700;
-    font-size: font-size--parent(1.6);
+    font-size: $font-size--lg;
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: reverted this change and used `font-size--parent(1)` instead f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 16]
```diff
@@ -11,65 +14,119 @@ $admin-space-card-logo-size: 80px;
   flex-direction: column;
   border: none;
   border-radius: $border-radius--xl;
-  padding: font-size--root(1);
   gap: font-size--root(0.5);
+  max-height: 450px;
+  box-shadow: 0px 2px 4px rgba(58, 92, 144, 0.14),
+    0px 3px 4px rgba(58, 92, 144, 0.12), 0px 1px 5px rgba(58, 92, 144, 0.2);
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: there's not much about box-shadow styles in `constants`, so I went ahead and created a separate var for this one `$admin-ng-space-card-box-shadow` f9043d4

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 17]
```diff
@@ -21,28 +24,71 @@ export interface AdminSpaceCardProps {
 
 export const AdminSpaceCard: React.FC<AdminSpaceCardProps> = ({ venue }) => {
   const backgroundStyle = useCss({
-    "background-image": `url(${venue.mapBackgroundImageUrl})`,
+    background: `linear-gradient(
+        rgba(25, 24, 26, 0.8),
+        rgba(25, 24, 26, 0.8)
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:41]: since I add this bg via tsx and not scss I'm not entirely sure how to use a variable in this case and if we should use it at all at this point

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 18]
```diff
@@ -11,65 +16,110 @@ $admin-space-card-logo-size: 80px;
...
   border-radius: $border-radius--xl;
-  padding: font-size--root(1);
   gap: font-size--root(0.5);
-
-  .AdminShowcaseSubTitle {
-    font-weight: $font-weight--700;
-    font-size: font-size--parent(1.6);
-    color: $content--over;
-  }
+  max-height: 450px;
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss:20]: any relation to `$admin-space-card-bg-height`? maybe 4:3 ratio? `3 * $admin-space-card-bg-height`?

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss [hunk 19]
```diff
@@ -11,65 +16,110 @@ $admin-space-card-logo-size: 80px;
...
 
   &__logo {
-    @include square-size($admin-space-card-logo-size);
+    @include square-size($admin-space-card-logo-size--lg);
+    margin-top: $spacing--lg;
     border-radius: $border-radius--max;
     background-size: cover;
     background-color: $content--under;
-    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.4);
+    box-shadow: 0 $spacing--xs $spacing--md opaque-black(0.4);
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss:112]: > // @deprecated please use var(--content--under-Xa) where X is the opacity e.g. var(--content--over-20a)

https://github.com/sparkletown/sparkle/blob/staging/src/scss/constants.scss#L42-L50

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (goran-peoski-work) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:41]: It is a static style.

My guess is it be applied as a background in SCSS that will get overridden by `background-image` set here. I think `background-image: linear-gradient()` should also be valid. But can't spend too much time tinkering with this.

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 20]
```diff
@@ -20,29 +30,85 @@ export interface AdminSpaceCardProps {
...
 
 export const AdminSpaceCard: React.FC<AdminSpaceCardProps> = ({ venue }) => {
+  const [validBannerImageUrl] = useValidImage(
+    venue?.config?.landingPageConfig.bannerImageUrl,
+    DEFAULT_VENUE_BANNER_COLOR
+  );
+
   const backgroundStyle = useCss({
-    "background-image": `url(${venue.mapBackgroundImageUrl})`,
+    background: `linear-gradient(
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:39]:
```suggestion
    // @debt There should be a way to move the rgba() functions to SCSS and use $color-constant instead of literals
    background: `linear-gradient(
```

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 21]
```diff
@@ -20,29 +30,85 @@ export interface AdminSpaceCardProps {
...
+            to={venueInsideUrl(venue.slug)}
+            target="_blank"
+            rel="noopener noreferer"
+          >
+            {"Visit"}
+            <FontAwesomeIcon
+              className="AdminSpaceCard__link-icon"
+              icon={faExternalLinkAlt}
+            />
+          </Link>
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:71]: [NOTE]

`ButtonNG` has also the ability to create external links:

```tsx
      <ButtonNG
        isLink
        newTab
        linkTo={venueInsideUrl(venue.slug)}
      >
```

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx [hunk 22]
```diff
@@ -20,29 +30,85 @@ export interface AdminSpaceCardProps {
...
-        <AdminShowcaseSubTitle>{venue.name}</AdminShowcaseSubTitle>
+      <div className={backgroundClasses}>
+        <div className="AdminSpaceCard__bg-container">
+          <Link
+            className="AdminSpaceCard__link"
+            to={venueInsideUrl(venue.slug)}
+            target="_blank"
+            rel="noopener noreferer"
+          >
+            {"Visit"}
```
Comment (goran-peoski-work) [inline src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:66]:
```suggestion
            Visit
```

File: src/scss/constants.scss (hunk 5 above)
Comment (goran-peoski-work) [reply src/scss/constants.scss:147]: Naming is OK, was just curious if there was an issue of adding the `//` comments in the same line after `;` as to not disturb the sequence

File: src/components/organisms/AdminVenueView/components/AdminCardTitle/AdminCardTitle.scss [hunk 23]
```diff
@@ -0,0 +1,16 @@
+@import "scss/constants";
+
+$height: 1rem;
```
Comment (goran-peoski-work) [inline src/components/organisms/AdminVenueView/components/AdminCardTitle/AdminCardTitle.scss:3]: [NOTE]

Using functions like `font-size--parent(1)` or `font-size--root(1)` adds more semantics (makes it more obvious what the intent was.

**Review by goran-peoski-work — APPROVED** [review]
LGTM

Notes and questions added, but other than a `@debt` comment, the rest can be left for some other PR

File: src/components/organisms/AdminVenueView/components/AdminCardTitle/AdminCardTitle.scss (hunk 23 above)
Comment (alexey-petrov) [reply src/components/organisms/AdminVenueView/components/AdminCardTitle/AdminCardTitle.scss:3]: it's kinda confusing that a function which name starts with `font-size` is being applied to css attributes that relate to height (eg. `min-height`, `line-height`) prob we can figure out another name? not sure

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 22 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: done 0ab68ac

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 21 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx]: that `ButtonNG` even with a `isLink` attribute differs greatly from the design, so I didn't use it

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx (hunk 20 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.tsx:41]: done 0ab68ac

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss (hunk 19 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: done 0ab68ac

File: src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss (hunk 18 above)
Comment (alexey-petrov) [reply src/components/molecules/AdminSpaceCard/AdminSpaceCard.scss]: done 0ab68ac
